src/dale/logger.py

```diff
@@ -13,10 +14,44 @@
 def init_parser() -> ArgumentParser:
     parser = ArgumentParser(description="Explicit logging of a list of APDUs",
                             formatter_class=RawTextHelpFormatter)
-    parser.add_argument("apdu_file", metavar="APDU_FILE", help="The file containing the list of APDUs", type=Path)
+    parser.add_argument("apdu_file", metavar="APDU_FILE",
+                        help=("The file containing the list of APDUs"),
+                        type=Path)
+    parser.add_argument("--json", action="store_true", help="Indicates that the input file is a Ledger Live log")
+    parser.add_argument("--inverted", action="store_true",
+                        help="Indicates that the Ledger Live log lines should be inverted")
+
     return parser
 
 
+def pre_format_ledger_live_log(json_file: Path, invert_lines: bool) -> Path:
+    with json_file.open("r", encoding="utf8") as f:
+        entries = json.load(f)
+
+    # Extract the APDUs from the log
+    # get the "message" field from each entry where the "type" is "apdu"
+    apdus = list(map(lambda x: x["message"], filter(lambda x: x.get("type", "") == "apdu", entries)))
+
+    # Create a new file with the same name as the Ledger live
+    # input file but ending with "_APDU.txt"
+    output_file = json_file.with_name(json_file.stem + "_APDU.txt")
```

Comment:
Not a fan of creating files in the current repo like this
If we really want to create the files we need to create them in /tmp